src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-inline-feedback.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Feedback, FeedbackType, MANUAL_ASSESSMENT_IDENTIFIER } from 'app/entities/feedback.model';
+import { cloneDeep } from 'lodash';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+    selector: 'jhi-code-editor-tutor-assessment-inline-feedback',
+    templateUrl: './code-editor-tutor-assessment-inline-feedback.component.html',
+})
+export class CodeEditorTutorAssessmentInlineFeedbackComponent {
+    MANUAL = FeedbackType.MANUAL;
+    @Input()
+    get feedback(): Feedback {
+        return this._feedback;
+    }
+    set feedback(feedback: Feedback) {
+        this._feedback = feedback || new Feedback();
+        this.oldFeedback = cloneDeep(this.feedback);
+        this.editOnly = feedback ? true : false;
+    }
+    private _feedback: Feedback;
+    @Input()
+    fileName: string;
+    @Input()
+    codeLine: number;
+    @Input()
+    readOnly: boolean;
+    @Output()
+    onUpdateFeedback = new EventEmitter<Feedback>();
+    @Output()
+    onCancelFeedback = new EventEmitter<number>();
+    @Output()
+    onDeleteFeedback = new EventEmitter<Feedback>();
+    @Output()
+    onEditFeedback = new EventEmitter<number>();
+
+    editOnly: boolean;
+    oldFeedback: Feedback;
+    constructor(private translateService: TranslateService) {}
+
+    /**
+     * Updates the current feedback and sets props and emits the feedback to parent component
+     */
+    updateFeedback() {
+        this.feedback.type = this.MANUAL;
+        this.feedback.reference = `${MANUAL_ASSESSMENT_IDENTIFIER}_file:${this.fileName}_line:${this.codeLine}`;
```

Comment:
For the future, it would be nice, if tutors can also select certain characters of a line. I think this could be extended using `file:${this.fileName}_line:${this.codeLine}_characters:{start}:{end}`

Reply:
It would be more important that `fileName` includes the path, in the (unlikely) case that there are two files with the same name

Reply:
The reference contains the whole path of the file. I renamed the variable to have a consistent name throughout the components. :)